web/src/sidebar_ui.js

```diff
@@ -19,3 +21,31 @@ export function show_streamlist_sidebar() {
 export function hide_streamlist_sidebar() {
     $(".app-main .column-left").removeClass("expanded");
 }
+
+$("body").on("click", ".login_button", (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    window.location.href = hash_util.build_login_link();
+});
+
+$("#userlist-toggle-button").on("click", (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    const sidebarHidden = !$(".app-main .column-right").hasClass("expanded");
+    popovers.hide_all();
+    if (sidebarHidden) {
+        show_userlist_sidebar();
+    }
+});
+
+$("#streamlist-toggle-button").on("click", (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    const sidebarHidden = !$(".app-main .column-left").hasClass("expanded");
+    popovers.hide_all();
+    if (sidebarHidden) {
+        show_streamlist_sidebar();
+    }
+});
```

Comment:
These should be in an `initialize` method -- we basically never want to run code at top-level (at import time) since it's problematic for writing tests and predictability about how a module behaves.

Reply:
Done